### PR TITLE
Yako Workstation: Scaffolding for workstation on project page

### PR DESCRIPTION
**Pull request by danielyaa5**
Created table component, table reducer, and clips reducers. Bother reducers map to table component.

### CODE REVIEW

File: src/routes/Project/actions/ClipActionHandlers.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import ProjectConstants from '../constants';
+
+const ACTION_HANDLERS = {
+  [ProjectConstants.ADD_CLIP]: (state, action) => {
```
Comment (asg0451) [inline src/routes/Project/actions/ClipActionHandlers.js:4]: we're using babel right? I do like this syntax but it's definitely not widely supported.

File: src/routes/Project/actions/ClipActionHandlers.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    const id = action.payload.id;
+    const track = action.payload.start.y;
+    const startTime = action.payload.start.x;
+    const endTime = action.payload.end.x;
+    return { ...state,  [id]: { track, startTime, endTime } };
```
Comment (asg0451) [inline src/routes/Project/actions/ClipActionHandlers.js]: as i understand it, the state variable coming into this function is the subset of the state concerning clips only, correct?

File: src/routes/Project/actions/TableActionHandlers.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import ProjectConstants from '../constants';
+
+const ACTION_HANDLERS = {
+  [ProjectConstants.ADD_CLIP_TO_TRACK]: (state, action) => {
+    const id = action.payload.id;
+    const track = action.payload.start.y;
+    const startTime = action.payload.start.x;
+    const endTime = action.payload.end.x;
+    return { ...state,  [track]: { ...state[track], [startTime]: id, [endTime]: id } };
```
Comment (asg0451) [inline src/routes/Project/actions/TableActionHandlers.js:9]: this is kind of an odd way of organizing this, but it kind of makes sense. any particular reason you decided to do it this way?

File: src/routes/Project/components/Table.js [hunk 4]
```diff
@@ -1,11 +1,16 @@
-import React from 'react'
+import React from 'react';
 
 export const Table = (props) => (
```
Comment (asg0451) [inline src/routes/Project/components/Table.js:3]: where do these props come from? I see you instantiating Table in the file above as `<Table/>,` don't you need to give it props there or do i misunderstand.

Comment (asg0451) [reply src/routes/Project/components/Table.js:3]: also, how can the Table access the store? I'm not quite sure i understand that part.

File: src/routes/Project/containers/TableContainer.js [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
-
-}
+  addClipToTrack,
+  addClip
+};
 
 const mapStateToProps = (state) => ({
+  table: state.table,
+  clips: state.clips
+});
```
Comment (asg0451) [inline src/routes/Project/containers/TableContainer.js:15]: is this where the props come from? from the store?

File: src/routes/Project/index.js [hunk 6]
```diff
@@ -1,12 +1,21 @@
+import { injectReducer } from '../../store/reducers';
+
 export default (store) => ({
     path: 'projects/:id',
     getComponent(nextState, cb) {
```
Comment (asg0451) [inline src/routes/Project/index.js:5]: what does this function do?

File: src/routes/Projects/components/ProjectsView.js [hunk 7]
```diff
@@ -1,17 +1,15 @@
-import React from 'react'
+import React from 'react';
 
-import ProjectsList from './ProjectsList'
-import './ProjectsView.scss'
+import ProjectsList from './ProjectsList';
```
Comment (asg0451) [inline src/routes/Projects/components/ProjectsView.js:3]: where is this defined? or is it not written yet

File: src/routes/Project/actions/ClipActionHandlers.js (hunk 1 above)
Comment (danielyaa5) [reply src/routes/Project/actions/ClipActionHandlers.js:4]: babel, yes. You need babel for JSX transformation

File: src/routes/Project/actions/ClipActionHandlers.js (hunk 2 above)
Comment (danielyaa5) [reply src/routes/Project/actions/ClipActionHandlers.js]: Ya its the map of clips for the project, I just realized its missing audio attribute which will store a path to the audio file, most likely an s3 link

File: src/routes/Project/actions/TableActionHandlers.js (hunk 3 above)
Comment (danielyaa5) [reply src/routes/Project/actions/TableActionHandlers.js:9]: ya see the google drive sheet about data organization, basically its crucial for preventing clip overlap.

File: src/routes/Project/components/Table.js (hunk 4 above)
Comment (danielyaa5) [reply src/routes/Project/components/Table.js:3]: thats the magic of redux, they come from the connect function call in the TableContainer file. The state is mapped to the props with the mapStateToProps function.

File: src/routes/Project/containers/TableContainer.js (hunk 5 above)
Comment (danielyaa5) [reply src/routes/Project/containers/TableContainer.js:15]: yes

File: src/routes/Project/index.js (hunk 6 above)
Comment (danielyaa5) [reply src/routes/Project/index.js:5]: its used by the react router to route the projectview to the path, it also adds the reducers to the main reducer.

File: src/routes/Projects/components/ProjectsView.js (hunk 7 above)
Comment (danielyaa5) [reply src/routes/Projects/components/ProjectsView.js:3]: projects/components/projectlist

**Pull request merged**